File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDONoticeMessageStateParamModel.kt
//
//  IDONoticeMessageStateParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.FieldNamingPolicy
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import java.io.Serializable
import com.google.gson.annotations.SerializedName

///


open class IDONoticeMessageStateParamModel(
    itemsNum: Int,
    operat: Int,
    allOnOff: Int,
    allSendNum: Int,
    allNotifyState: Int,
    nowSendIndex: Int,
    items: List<IDONoticeMessageStateItemItem>
) : IDOBaseModel {

    @SerializedName("version")
    private var version: Int = 1

    @SerializedName("items_num")
    var itemsNum: Int = itemsNum

    /// Operation 1: Add  2: Modify 3: Get and Query
    @SerializedName("operat")
    var operat: Int = operat

    /// Add and modify only
    /// Overall notification switch
    /// 1: Enable all notifications
    /// 0: Disable all notifications
    @SerializedName("all_on_off")
    var allOnOff: Int = allOnOff

    @SerializedName("all_notify_state")
    var allNotifyState: Int = allNotifyState

    /// Total number of packets sent
    /// For sending more than 100 packets in multiple parts
    /// all_send_num = now_send_index for completion of sending
    @SerializedName("all_send_num")
    var allSendNum: Int = allSendNum

    /// Current sequence of sending
    @SerializedName("now_send_index")
    var nowSendIndex: Int = nowSendIndex

    /// Message details
    /// Collection of evt_type, notify_state, and pic_flag
    @SerializedName("items")
    var items: List<IDONoticeMessageStateItemItem> = items

    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}


// MARK: - IDONoticeMessageStateItemItem
open class IDONoticeMessageStateItemItem(evtType: Int, notifyState: Int, picFlag: Int) :
    Serializable {
    /// Event type
    @SerializedName("evt_type")
    var evtType: Int = evtType

    /// Notification status
    /// 1: Allow notifications
    /// 2: Silent notifications
    /// 3: Close notifications
    @SerializedName("notify_state")
    var notifyState: Int = notifyState

    /// Applies when replying, set this parameter to 0
    /// 0: Invalid
    /// 1: Download corresponding image
    /// 2: No corresponding image
    @SerializedName("pic_flag")
    var picFlag: Int = picFlag

}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDONoticeMessageStateParamModel.swift
//
//  IDONoticeMessageStateParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/19.
//

import Foundation

@objcMembers
public class IDONoticeMessageStateParamModel: NSObject, IDOBaseModel {
    private var version: Int
    private var itemsNum: Int
    /// Operation 1: Add  2: Modify 3: Get and Query
    public var operat: Int
    /// Add and modify only
    /// Overall notification switch
    /// 1: Enable all notifications
    /// 0: Disable all notifications
    public var allOnOff: Int
    /// Total number of packets sent
    /// For sending more than 100 packets in multiple parts
    /// all_send_num = now_send_index for completion of sending
    public var allSendNum: Int
    /// Current sequence of sending
    public var nowSendIndex: Int
    /// Message details
    /// Collection of evt_type, notify_state, and pic_flag
    public var items: [IDONoticeMessageStateItem]
    
    enum CodingKeys: String, CodingKey {
        case version = "version"
        case itemsNum = "items_num"
        case operat = "operat"
        case allOnOff = "all_on_off"
        case allSendNum = "all_send_num"
        case nowSendIndex = "now_send_index"
        case items = "items"
    }
    
    public init(operat: Int, allOnOff: Int, allSendNum: Int, nowSendIndex: Int, items: [IDONoticeMessageStateItem]) {
        self.version = 0
        self.itemsNum = items.count
        self.operat = operat
        self.allOnOff = allOnOff
        self.allSendNum = allSendNum
        self.nowSendIndex = nowSendIndex
        self.items = items
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

// MARK: - IDONoticeMessageStateItem
@objcMembers
public class IDONoticeMessageStateItem: NSObject, Codable {
    /// Event type
    public var evtType: Int
    /// Notification status
    /// 1: Allow notifications
    /// 2: Silent notifications
    /// 3: Close notifications
    public var notifyState: Int
    /// Applies when replying, set this parameter to 0
    /// 0: Invalid
    /// 1: Download corresponding image
    /// 2: No corresponding image
    public var picFlag: Int
    
    enum CodingKeys: String, CodingKey {
        case evtType = "evt_type"
        case notifyState = "notify_state"
        case picFlag = "pic_flag"
    }
    
    public init(evtType: Int, notifyState: Int, picFlag: Int) {
        self.evtType = evtType
        self.notifyState = notifyState
        self.picFlag = picFlag
    }
}
